rewrite-core/src/main/java/org/openrewrite/internal/ExceptionUtils.java

```diff
@@ -43,4 +46,26 @@ public static String sanitizeStackTrace(Throwable t, Class<?> until) {
         }
         return sanitized.toString();
     }
+
+    public static boolean containsCircularReferences(Throwable exception) {
+        List<Throwable> causes = new LinkedList<Throwable>();
```

Comment:
Detail: It would be more efficient with a `Set`. Specifically, I suggest we use `Collections.newSetFromMap(new IdentityHashMap<>())`.

Reply:
Thanks! I did not know about the `IdentityHashMap`